### PR TITLE
Extend ability files to have any number of fields

### PR DESCRIPTION
**Pull request by garunagiri**
The changes here allow ability files to have arbitrary fields that get read and stored in abilities. This allows other plugins to extend per-ability logic based on values specific to individual abilities.

### CODE REVIEW

**Comment by privateducky** [comment]
@khyberspache think there's too much overlap between this and the ability hooks? @garunagiri can you check with @khyberspache to see if the hooks would work for your use case?

**Comment by garunagiri** [comment]
@khyberspache and I just chatted about this. It seems to us like the hook capability solves different problems than the particular use case that inspired this change. The use case doesn't require post-processing of abilities, but rather just extra information about each ability. Is there something I'm not seeing about the hook capability that could be used here?

**Comment by khyberspache** [comment]
@privateducky He's adding an additional field to the actual YAML file.  It could in the future be used with the HOOKS to add power to the planner based upon the field.

**Comment by wbooth** [comment]
Like the direction. what is your plan for displaying ability fields on the UI and in reports? Do you envision all additional attributes being editable (from the UI)?

**Comment by garunagiri** [comment]
> Like the direction. what is your plan for displaying ability fields on the UI and in reports? Do you envision all additional attributes being editable (from the UI)?

I do see all additional attributes being editable from the UI. I also plan on allowing them to be deleted and for new ones to be added from the front end as well. I'll put these in in a later (soon) PR.

File: app/service/data_svc.py [hunk 1]
```diff
@@ -135,36 +135,40 @@ async def _load_abilities(self, plugin):
...
+                    technique_name = ab.get('technique', dict()).get('name')
+                    technique_id = ab.pop('technique', dict()).get('attack_id')
+                    ability_id = ab.pop('id', None)
+                    tactic = ab.pop('tactic', None)
+                    description = ab.pop('description', '')
+                    ability_name = ab.pop('name', '')
+                    privilege = ab.pop('privilege', None)
+                    repeatable = ab.pop('repeatable', False)
+                    requirements = ab.pop('requirements', [])
+                    for platforms, executors in ab.pop('platforms', []).items():
```
Comment (wbooth) [inline app/service/data_svc.py:148]: default to dict not list

File: app/service/data_svc.py [hunk 2]
```diff
@@ -176,10 +180,9 @@ async def _update_extensions(self, ability):
             ab.technique_name = ability.technique_name
             await self.store(ab)
 
-    async def _classify(self, ability):
-        if 'buckets' in ability:
-            return ability['buckets'].lower()
-        return [ability['tactic'].lower()]
+    @staticmethod
+    async def _classify(ability, tactic):
+        return ability.pop('buckets', tactic).lower()
```
Comment (wbooth) [inline app/service/data_svc.py:185]: is buckets not a list?

**Pull request merged**